File: backend/company/src/main/java/com/sgtrain/company/service/CompanyService.java
package com.sgtrain.company.service;

import java.util.Optional;

import com.sgtrain.company.dto.CompanyDto;

public interface CompanyService {
	
	public boolean createCompany(CompanyDto companyDto);
	public Iterable<CompanyDto> getCompany();
	public Optional<CompanyDto> getCompanyById(String id);
	public boolean deleteCompany(String id);
	public CompanyDto editCompany(String id, CompanyDto companyDto);

}

File: frontend/stock-app/src/app/services/company-service/company.service.ts
import { HttpClient, HttpHeaders } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { CompanyDto } from 'src/app/models/company/company-dto';

@Injectable({
  providedIn: 'root'
})
export class CompanyService {

  baseUrl ='http://localhost:8080/company/display';
  createCompanyUrl ='http://localhost:9002/company/create';
  deleteCompanyUrl ='http://localhost:9002/company/remove';

  constructor(private httpClient: HttpClient) { }

  getCompanyList(): Observable<CompanyDto[]> {
    console.log( this.httpClient.get<CompanyDto[]>(`${this.baseUrl}`));
    return this.httpClient.get<CompanyDto[]>(`${this.baseUrl}`);
  }

  createCompany(companyDto: CompanyDto): Observable<boolean> {
    const httpOptions = {
      headers: new HttpHeaders({
        'Content-Type':  'application/json'
      })
    };
    return this.httpClient.post<boolean>(`${this.createCompanyUrl}`, companyDto, httpOptions);
  }

  deleteCompany(id: String): Observable<Object> {
    return this.httpClient.delete(`${this.deleteCompanyUrl}/${id}`);
  }


}
